test/runTest.ts

```diff
@@ -1,19 +1,32 @@
+import * as cp from 'child_process';
 import * as path from 'path';
-
-import { runTests } from 'vscode-test';
+import {
+  downloadAndUnzipVSCode,
+  resolveCliPathFromVSCodeExecutablePath,
+  runTests
+} from 'vscode-test';
 
 async function main() {
     try {
         // The folder containing the Extension Manifest package.json
         // Passed to `--extensionDevelopmentPath`
         const extensionDevelopmentPath = path.resolve(__dirname, '../../');
-
         // The path to the extension test script
         // Passed to --extensionTestsPath
         const extensionTestsPath = path.resolve(__dirname, './');
-
+        const vscodeExecutablePath = await downloadAndUnzipVSCode('1.50.0');
+        const cliPath = resolveCliPathFromVSCodeExecutablePath(vscodeExecutablePath)
+        // Use cp.spawn / cp.exec for custom setup
+        cp.spawnSync(cliPath, ['--install-extension', 'redhat.vscode-commons'], {
```

Comment:
We shouldn't do this.

Reply:
Seems legit to me : https://github.com/microsoft/vscode-java-test/blob/master/test/index.ts#L17

Reply:
> Seems legit to me : https://github.com/microsoft/vscode-java-test/blob/master/test/index.ts#L17
@fbricon, to be specific, my worry is that we shouldn't endup having hard dependency on vscode-commons. How to ensure the extension still works without having a dependency on vscode-commons.?

Reply:
We need a hard dependency on vscode-commons, else it'll never be installed automatically and you'll never get telemetry

Reply:
Ack!